src/App.tsx

```diff
@@ -1,24 +1,239 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import * as postService from './api/todos';
+import { getFilteredTodos } from './utils/functions';
+import { TodoItem } from './components/TodoItem';
+import { TodoFilter } from './components/TodoFilter';
+import { Header } from './components/Header';
+import { ErrorMessage } from './components/ErrorMessage';
+import { TempTodo } from './components/TempTodo';
+import { Status } from './types/Status';
+import { ErrorMessages } from './utils/ErrorMessages';
 
-const USER_ID = 0;
+const USER_ID = 11457;
+const initialTodos: Todo[] = [];
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[]>(initialTodos);
+  const [filterBy, setFilterBy] = useState(Status.All);
+  const [errorMessage, setErrorMessage] = useState(ErrorMessages.NoError);
+  const [title, setTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingId, setLoadingId] = useState<number[]>([]);
+  const [isLoaderActive, setIsLoaderActive] = useState(false);
+  const [isToggled, setIsToggled] = useState(false);
+
+  const filteredTodos = getFilteredTodos(todoList, filterBy);
+
+  useEffect(() => {
+    postService.getTodos(USER_ID)
+      .then(setTodoList)
+      .catch(() => {
+        setErrorMessage(ErrorMessages.LoadError);
+      });
+    const timerId = setInterval(() => {
+      setErrorMessage(ErrorMessages.NoError);
+    }, 3000);
+
+    return () => {
+      clearInterval(timerId);
+    };
+  }, []);
+
+  const allTodosId = todoList.map(({ id }) => id);
+
+  const completedTodosId = todoList
+    .reduce((total: number[], current) => (current.completed
+      ? [...total, current.id]
+      : total), []);
+
+  const hasCompletedTodosCount = !!completedTodosId.length;
+
+  const activeTodosCount = todoList
+    .filter(({ completed }) => completed === false).length;
+
+  const handleDeleteTodo = (todoId: number) => {
+    setLoadingId(prevTodoIds => [...prevTodoIds, todoId]);
+    setIsLoaderActive(true);
+
+    return postService.deleteTodo(todoId)
+      .then(() => {
+        setTodoList(currentList => currentList
+          .filter(todo => todo.id !== todoId));
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessages.DeleteError);
+      })
+      .finally(() => {
+        setLoadingId([]);
+        setIsLoaderActive(false);
+      });
+  };
+
+  const handleClearCompleted = () => {
+    completedTodosId
+      .forEach((id) => {
+        handleDeleteTodo(id);
+      });
+  };
+
+  const addNewTodo = (todo: Todo) => {
+    setTempTodo({
+      id: 0,
+      userId: USER_ID,
+      title: title.trim(),
+      completed: false,
+    });
+
+    setErrorMessage(ErrorMessages.NoError);
+    setLoadingId([0]);
+    setIsLoaderActive(true);
```

Comment:
Why do you need `setIsLoaderActive` if you store todos ids and should show loader if todo id in the current array?
Also, `0` as I understand temporary todo id, so maybe create a constant for this id?